### PR TITLE
Update/association UI update

### PR DESCRIPTION
**Pull request by ntissieres**

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt [hunk 1]
```diff
@@ -23,16 +18,12 @@ import com.github.swent.echo.data.model.Association
...
-    eventsFilter: List<Association>
 ) {
     LazyColumn(modifier = Modifier.testTag("association_list_screen")) {
         items(associationList.size) { index ->
             AssociationListElement(
                 association = associationList[index],
-                onRowClicked = { onRowClicked(it) },
-                onAssociationClicked = { onAssociationClicked(it) },
-                eventsFilter
+                onRowClicked = { onRowClicked(it) }
```
Comment (octogradiste) [inline app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt:26]: You could simply use `onRowClicked = onRowClicked`

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt [hunk 2]
```diff
@@ -43,39 +34,23 @@ fun AssociationListScreen(
...
-                        .testTag("selected_icon_${association.name}")
-                )
-            }
+            Text(
+                text = association.name,
+                textAlign = TextAlign.Center,
+                modifier =
+                    Modifier.padding(boxInsidePadding)
+                        .testTag("association_name_button_${association.name}")
+            )
```
Comment (octogradiste) [inline app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt:53]: You could move this inside `headlineContent` avoids having `headlineContent = {}`...

File: app/src/main/java/com/github/swent/echo/compose/components/Pager.kt [hunk 3]
```diff
@@ -33,8 +35,8 @@ import kotlinx.coroutines.launch
  */
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun Pager(content: List<Pair<String, @Composable () -> Unit>>) {
-    val pagerState = rememberPagerState(pageCount = { content.size })
+fun Pager(content: List<Pair<String, @Composable () -> Unit>>, initialPage: Int? = 0) {
```
Comment (octogradiste) [inline app/src/main/java/com/github/swent/echo/compose/components/Pager.kt:38]: You could use `initialPage: Int = 0` as you never need initial page to be null.

File: app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt [hunk 4]
```diff
@@ -109,41 +73,26 @@ constructor(
...
-            allEvents.filter {
-                _followedAssociations.value.contains(it.organizer) ||
-                    _committeeAssociations.value.contains(it.organizer)
-            }
-        } else {
-            allEvents.filter { _eventsFilter.value.contains(it.organizer) }
-        }
+    fun setCurrentAssociationPage(
+        association: Association,
+        initialPage: Int? = _initialPage.value
```
Comment (octogradiste) [inline app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt:89]: Here `initialPage` could also be of type `Int` and not `Int?`.

**Review by octogradiste — APPROVED** [review]
Thanks so much for the association UI improvement! The code is very clean and easy to read. I left some comments on stuff you could improve, but as it's rather minor I already approved. Up to you to make the changes or merge directly :)

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt (hunk 2 above)
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/association/AssociationListScreen.kt:53]: Right ! it's more logical

File: app/src/main/java/com/github/swent/echo/compose/components/Pager.kt (hunk 3 above)
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/components/Pager.kt:38]: I thought the ? was the way to allow a function to not take argument, I didn't know that we could do it this way ! Thanks for highlighting this ;)

**Review by octogradiste — APPROVED** [review]
LGTM!!

**Pull request merged**